### PR TITLE
Stream phantomjs output to console

### PR DESCRIPTION
**Pull request by haiiaaa**
Support for large buffers by using spawn instead of execFile (before it would throw "Error: stdout maxBuffer exceeded")

### CODE REVIEW

**Comment by jonkemp** [comment]
I think this looks better. I'll look at it more tonight. Does this replace #33? If so, just close that one please.

**Comment by jonkemp** [comment]
Question, I haven't tried it yet, but do you think this will run the tests faster?

**Comment by haiiaaa** [comment]
I don't think it will make any difference.

**Comment by jonkemp** [comment]
Ok, I love this change! It is much faster. I ran the test from this branch twice, once at 8 seconds and once at 10 seconds. I checked out master and ran the current tests which finished after 33 seconds. So, great job and much thanks!

My current plan is to merge this in and then publish it as a new version, 2.0.

**Comment by jonkemp** [comment]
Done, 5e9e7278e9a630d385731118ed35ba14a5e4c828.

**Comment by haiiaaa** [comment]
Nice! Thanks.

File: index.js [hunk 1]
```diff
@@ -50,51 +52,40 @@ module.exports = function (filepath, options, callback) {
...
             }
         }
 
-        if (stderr) {
-            console.log(stderr);
-        }
+        if (out.result) {
+            message = 'Took ' + out.result.runtime + ' ms to run ' + out.result.total + ' tests. ' + out.result.passed + ' passed, ' + out.result.failed + ' failed.';
 
-        if (err) {
```
Comment (jonkemp) [inline index.js]: @haiiaaa question, it appears there is no equivalent in `childProcess.spawn` to this error in the callback of `childProcess.execFile`. Test failures will cause that error to be thrown and then you can show different messages if there are test failures versus no test failures. Is there another way to achieve the same thing with spawn?

Comment (haiiaaa) [reply index.js]: According to the documentation this should be possible:

    proc.on('error', (err) => {
      console.log(err);
    });

Comment (haiiaaa) [reply index.js]: Yeah, and in es5 that would be 

    proc.on('error', function (err) {
      console.log(err);
    });

Would you like a PR?

Comment (jonkemp) [reply index.js]: Yeah, I tried that and it does not do what I am looking for. It only errors in that case if the process fails to spawn somehow.

Comment (haiiaaa) [reply index.js]: Ok, now I see that here: https://nodejs.org/api/child_process.html#child_process_event_error

By test failures, do you mean failures in `out.exceptions` or do errors thrown from the javascript code?

Comment (jonkemp) [reply index.js]: I think the qunit runner throws a javascript error if the test fails.

Comment (haiiaaa) [reply index.js]: I guess you want the following as output to the console if any test fails, right? I have not been able to achieve it with spawn.

    { Error: Command failed: C:\ProgramData\chocolatey\bin\phantomjs.EXE C:\dev\github\test\node_modules\qunit-phantomjs-runner\runner-json.js ./tests.htm 120
    
        at ChildProcess.exithandler (child_process.js:270:12)
        at emitTwo (events.js:130:20)
        at ChildProcess.emit (events.js:213:7)
        at maybeClose (internal/child_process.js:927:16)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
      killed: false,
      code: 1,
      signal: null,
      cmd: 'C:\\ProgramData\\chocolatey\\bin\\phantomjs.EXE C:\\dev\\github\\test\\node_modules\\qunit-phantomjs-runner\\runner-json.js ./tests.htm 120' }

If this is a showstopper it is also possible to stream with execFile but then the buffer would have to be specified manually for large buffers.

     buffer = options.buffer || 200; // default value
     proc = childProcess.execFile(binPath, childArgs, { maxBuffer: buffer*1024 }, function (err){
        if (err) {
            console.log(err);
        }
    });

Comment (jonkemp) [reply index.js]: I found a solution. Just wondering if you knew of anything. Thanks.

**Pull request merged**